plugin-parser/mlops-plugin-parser.cc

```diff
@@ -179,45 +375,100 @@ _parse_json (JsonNode *node, mlsvc_json_type_e json_type, const gchar *app_info)
 
           if (path_len == 0U) {
             _E ("Failed to get path from MLSVC_JSON_RESOURCE.");
-            continue;
+            return FALSE;
           }
 
-          for (guint pidx = 0; pidx < path_len; pidx++) {
-            const gchar *path = NULL;
-
-            if (path_array)
-              path = json_array_get_string_element (path_array, pidx);
-            else
-              path = json_node_get_string (path_node);
-
-            if (!path) {
-              _E ("Failed to get path at '%d'th of '%s' from MLSVC_JSON_RESOURCE.",
-                  pidx, name);
-              continue;
+          if (event == MLSVC_PKGMGR_MDPARSER_PLUGIN_EVENT_TYPE_INSTALL) {
+            for (guint pidx = 0; pidx < path_len; pidx++) {
+              const gchar *path = NULL;
+
+              if (path_array)
+                path = json_array_get_string_element (path_array, pidx);
+              else
+                path = json_node_get_string (path_node);
+
+              if (!path) {
+                _E ("Failed to get path at '%d'th of '%s' from MLSVC_JSON_RESOURCE.",
+                    pidx, name);
+                return FALSE;
+              }
+
+              ret = ml_agent_resource_add (
+                  name, path, desc ? desc : "", app_info ? app_info : "");
+
+              if (ret == 0)
+                _I ("The resource at '%d'th of name '%s' is registered.", pidx, name);
+              else {
+                _E ("Failed to register the resource with name '%s'.", name);
+                return FALSE;
+              }
+            }
+          } else if (event == MLSVC_PKGMGR_MDPARSER_PLUGIN_EVENT_TYPE_UNINSTALL) {
+            char *res_info = NULL;
```

Comment:
ditto g_autofree gchar *res_info = NULL;

Reply:
thanks, i update it.